src/main/java/de/tum/in/www1/artemis/domain/exam/StudentExam.java

```diff
@@ -0,0 +1,154 @@
+package de.tum.in.www1.artemis.domain.exam;
+
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.JoinTable;
+import javax.persistence.ManyToMany;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+
+import de.tum.in.www1.artemis.domain.AbstractAuditingEntity;
+import de.tum.in.www1.artemis.domain.Exercise;
+import de.tum.in.www1.artemis.domain.User;
+
+@Entity
+@Table(name = "student_exam")
+public class StudentExam extends AbstractAuditingEntity {
+
+    // region CONSTRUCTORS
+    // -----------------------------------------------------------------------------------------------------------------
+
+    // no arg constructor required for jpa
+    public StudentExam() {
+    }
+
+    public StudentExam(Long id, Exam exam, User user, Set<Exercise> exercises) {
+        this.id = id;
+        this.exam = exam;
+        this.user = user;
+        this.exercises = exercises;
+    }
+
+    // -----------------------------------------------------------------------------------------------------------------
+    // endregion
+
+    // region BASIC PROPERTIES
+    // -----------------------------------------------------------------------------------------------------------------
+
+    @Id
+    @Column(name = "id")
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    // -----------------------------------------------------------------------------------------------------------------
+    // endregion
+
+    // region RELATIONSHIPS
+    // -----------------------------------------------------------------------------------------------------------------
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "exam_id")
+    private Exam exam;
+
+    public Exam getExam() {
+        return exam;
+    }
+
+    public void setExam(Exam exam) {
+        if (this.exam != null) {
+            this.exam.removeStudentExam(this);
```

Comment:
I would be careful with additional logic like this. Please have a look how this is handled in other places in our code, so that we do not break the consistency

Reply:
We simply do not implement the additional logic in the entities. E.g. in `Submission`:

```
public void setParticipation(Participation participation) {
        this.participation = participation;
}
```

Therefore I'll remove it in the new entities.